testsuite/pytests/test_eprop_plasticity.py

```diff
@@ -0,0 +1,738 @@
+# -*- coding: utf-8 -*-
+#
+# test_eprop_plasticity.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test functionality of e-prop plasticity.
+"""
+
+import nest
+import numpy as np
+import pytest
+
+
+def test_ConnectNeuronsWithEpropSynapse():
+    """Ensures that the restriction to supported neuron models works."""
+
+    nest.set_verbosity("M_WARNING")
+
+    supported_source_models = [
+        "eprop_iaf_psc_delta",
+        "eprop_iaf_psc_delta_adapt",
+    ]
+    supported_target_models = supported_source_models + ["eprop_readout"]
+
+    # Connect supported models with e-prop synapse
+
+    for nms in supported_source_models:
+        for nmt in supported_target_models:
+            nest.ResetKernel()
+
+            ns = nest.Create(nms, 1)
+            nt = nest.Create(nmt, 1)
+
+            nest.Connect(ns, nt, {"rule": "all_to_all"}, {"synapse_model": "eprop_synapse"})
+
+    # Ensure that connecting not supported models fails
+
+    for nm in [n for n in nest.node_models if n not in supported_target_models]:
+        nest.ResetKernel()
+
+        n = nest.Create(nm, 2)
+
+        with pytest.raises(nest.kernel.NESTError):
+            nest.Connect(n, n, {"rule": "all_to_all"}, {"synapse_model": "eprop_synapse"})
+
+
+def test_EpropRegression():
```

Comment:
This test and the following one are very long and difficult to understand. Can you provide some documentation about the logic of the tests?

Reply:
Since these tests are very close in structure to the examples, it would make sense to just provide a reference from the test to the example instead of repeating explanations here. It would also be good to include a link to the TensorFlow reference implementation.

Reply:
Fixed in https://github.com/nest/nest-simulator/pull/2867/commits/2432a91c3d5c07ab51f21674b8bc73df4bbdd5ac.